schema/gen/go/externUtil.go

```diff
@@ -0,0 +1,40 @@
+package gengo
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"path"
+	"strings"
+)
+
+// getExternTypes provides a mapping of all types defined in the destination package.
+// It is used by generate to not duplicate defined types to allow overriding of types.
+func getExternTypes(pth string) (map[string]struct{}, error) {
+	set := token.NewFileSet()
+	packs, err := parser.ParseDir(set, pth, nil, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	types := make(map[string]struct{})
+	for _, pack := range packs {
+		for fname, f := range pack.Files {
+			if strings.HasPrefix(path.Base(fname), "ipldsch_") {
```

Comment:
I know this is a generator and the performance isn't the most important thing, but parsing all files to then only look at a small minority is a bit wasteful :)

It would be significantly less work to use globbing to find the `ipldsch_*.go` files we're interested in, and then parse and inspect those.

Reply:
This is excluding the three `ipldsch_*.go` files and looking at all others to find types that are defined not in generated code. (this also addresses your other comment of "is it stable over multiple runs")

Reply:
:facepalm: I read the logic backwards. I guess my point here is still valid, though now it's just "you can avoid reading+parsing these 3-4 files".